### ISSUE TITLE
Make [DoesNotReturn] works for constructor

### ISSUE THREAD

**Issue by GF-Huang** [dotnet/roslyn#51860]
**Brief description:**
In WPF, a parameterless constructor is usually used for design time. It's need something attribute to tell analyzer do not check it.

**Languages applicable:**
C#

**Code example that the analyzer should report:**
```cs
public class ViewModel {
    private readonly IXxxService _service;
    // for runtime
    public ViewModel(IXxxService service) {
        _service = service;
    }
    // for design time, should never run at run time
    [NeverRun] // need something like this for preventing `CS8618` of `_service`
    public ViewModel() {
    }
}
```

**Additional information:**

Any more additional information you would like to add.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by huoyaoyuan** [comment]
Supressing CS8618 manually looks more correct here since it's intentional design, and you should also take care that null fields won't crash designer.

**Comment by RikkiGibson** [comment]
It feels like the ask here is to provide a convenient way to suppress initialization warnings in constructors. I think we might end up delivering that as part of the required properties feature. dotnet/csharplang#3630
